File: app-examples/example-jv-03adapters/src/main/java/foo/bar/example/foreadapters/feature/playlist/RandomTrackGeneratorUtil.java
package foo.bar.example.foreadapters.feature.playlist;

import java.util.Random;

import foo.bar.example.foreadapters.R;

public class RandomTrackGeneratorUtil {

    private static Random random = new Random();

    private static int[] colours = {
            R.color.pastel1,
            R.color.pastel2,
            R.color.pastel3,
            R.color.pastel4,
            R.color.pastel5
    };

    public static int generateRandomColourResource() {
        return randomInt(colours);
    }

    public static int randomInt(int[] intArray) {
        return intArray[random.nextInt(intArray.length - 1)];
    }
}

File: example-kt-03adapters/src/main/java/foo/bar/example/foreadapterskt/feature/playlist/RandomTrackGeneratorUtil.kt
package foo.bar.example.foreadapterskt.feature.playlist

import foo.bar.example.foreadapterskt.R
import java.util.Random


object RandomTrackGeneratorUtil {

    private val random = Random()

    private val colours = intArrayOf(R.color.pastel1, R.color.pastel2, R.color.pastel3, R.color.pastel4, R.color.pastel5)

    fun generateRandomColourResource(): Int {
        return randomInt(colours)
    }

    fun randomInt(intArray: IntArray): Int {
        return intArray[random.nextInt(intArray.size - 1)]
    }

}
